kaiax/gasless/impl/execution.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2025 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package impl
+
+import (
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/kaiax"
+)
+
+var _ kaiax.ExecutionModule = (*GaslessModule)(nil)
+
+func (g *GaslessModule) PostInsertBlock(block *types.Block) error {
+	return g.updateSupportedTokens(block.Number())
```

Comment:
We should also update `g.swapRouter` address by reading SystemRegistry every block.
If that's too many EVM call, we can add a function in `Multicall.sol` like [this](https://github.com/kaiachain/kaia/blob/5c596ffad/contracts/contracts/system_contracts/multicall/MultiCallContract.sol#L110-L113). What do you think?

Reply:
Using `Multicall.sol` would be better considering that other modules also execute contracts.

Reply:
fixed at https://github.com/kaiachain/kaia/pull/286/commits/86b55a089791e85e9a28df093b317f80e2bcf663, https://github.com/kaiachain/kaia/pull/286/commits/0a7b02a86475f59924de7323e89d40a009b28454, and https://github.com/kaiachain/kaia/pull/286/commits/8710286c59c6b35f967f7a09054f678bcdf29879.
I'll add contract test later.